src/test/java/com/google/sps/servlets/CaptionTest.java

```diff
@@ -0,0 +1,109 @@
+package com.google.sps.servlets;
+
+import static org.mockito.ArgumentMatchers.anyString;
+import static org.mockito.ArgumentMatchers.contains;
+import static org.mockito.ArgumentMatchers.eq;
+import static org.mockito.AdditionalMatchers.not;
+import static org.mockito.Mockito.when;
+
+import java.io.ByteArrayInputStream;
+import java.io.IOException;
+import java.io.InputStream;
+import org.junit.Assert;
+import org.junit.Before;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.junit.runners.JUnit4;
+import org.mockito.Mock;
+import org.mockito.MockitoAnnotations;
+
+@RunWith(JUnit4.class)
+public final class CaptionTest {
+  @Mock
+  private CaptionService captionServiceMock;
+
+  final private String PROPER_XML = "<?xml version=\"1.0\" encoding=\"utf-8\" ?><transcript>"
+      + "<text start=\"4.029\" dur=\"5.401\">line1</text>"
+      + "<text start=\"9.43\" dur=\"5.06\">line2</text>"
+      + "<text start=\"14.49\" dur=\"5.03\">line3</text>" + "</transcript>";
+
+  final private String PROPER_XML_CAPTION = "line1 line2 line3";
+
+  final private String BAD_XML = "<?xml version=\"1.0\" encoding=\"utf-8\" ?><transcript>"
+      + "<text start=\"4.029\" dur=\"5.401\">line1</text>"
+      + "<text start=\"9.43\" dur=\"5.06\">line2</text>"
+      + "<text start=\"14.49\" dur=\"5.03\">line3</text>";
+
+  final private String FAILED_CAPTION = null;
+
+  final private String VIDEO_ID = "sampleId";
+
+  final private String LANG_EN = "en";
+
+  final private String LANG_EN_GB = "en-GB";
+
+  @Before
+  public void init() {
+    MockitoAnnotations.initMocks(this);
+  }
+
+  @Test
+  public void properXml() throws IOException {
+    InputStream stream = new ByteArrayInputStream(PROPER_XML.getBytes());
+    when(captionServiceMock.fetchStream(eq(VIDEO_ID), contains(LANG_EN))).thenReturn(stream);
+
+    String actual = new Caption(captionServiceMock).getCaptionFromId(VIDEO_ID);
+    String expected = PROPER_XML_CAPTION;
+    Assert.assertEquals(expected, actual);
+  }
+
+  @Test
+  public void badXml() throws IOException {
```

Comment:
Use more descriptive names for your tests so it's obvious what is being tested and how. ToTT #339, Writing Descriptive Test Names, goes more into detail about it.

Reply:
Done. I like how you named your tests!